server/src/main/java/io/druid/segment/realtime/RealtimeManager.java

```diff
@@ -378,14 +358,19 @@ private void runFirehoseV2(FirehoseV2 firehose)
           metrics.incrementUnparseable();
```

Comment:
Isn't this catch block just the illustration of the above? If Exception here is an `InterruptedException`, it will be logged, but the interruption status is cleared and `haveRow` is still `true`.